### ISSUE TITLE
service: disable non-existing enabled systemd services

### ISSUE THREAD

**Issue by sanchox** [ansible/ansible#24942]
_From @prupert on September 15, 2016 13:6_

##### ISSUE TYPE
- Bug Report
##### COMPONENT NAME

service module
##### ANSIBLE VERSION

```
ansible 2.1.1.0
```
##### OS / ENVIRONMENT

CentOS 7
##### SUMMARY

When a systemd service is enabled (a symlink exists in `/etc/systemd/system/multi-user.target.wants`), but not in fact existing (the symlink is broken so to speak), the service module fails to disable the service. 
##### STEPS TO REPRODUCE
- Create a broken symbolic link `my_non_existing_service.service` in `/etc/systemd/system/multi-user.target.wants` on the target host with systemd. (This can happen if a previously enabled service `my_non_existing_service` gets removed.)
- Run a task that disables this service

```
- name: Ensure service is stopped and disabled.
  service: name=my_non_existing_service
           state=stopped
           enabled=false
```
##### EXPECTED RESULTS

I expect that the service `my_non_existing_service` will be disabled by having the symbolic link removed from `/etc/systemd/system/multi-user.target.wants`.

This is the same as when run `systemctl disable my_non_existing_service.service`, the output when running this command manually on the target host will be:

```
Removed symlink /etc/systemd/system/multi-user.target.wants/my_non_existing_service.service.
```
##### ACTUAL RESULTS

The symbolic link to `my_non_existing_service` keeps being there, thus leaving a non-existing service "enabled".  
The error output (redacted) is as follows:

```
failed: [target-host-name] ( {"failed": true, "invocation": {"module_args": {"arguments": "", "enabled": false, "name": "my_non_existing_service", "pattern": null, "runlevel": "default", "sleep": null, "state": "stopped"}, "module_name": "service"}, "msg": "systemd could not find the requested service \"'my_non_existing_service'\": "}
```

_Copied from original issue: ansible/ansible-modules-core#4864_

**Comment by sanchox** [comment]
_From @ansibot on September 15, 2016 13:23_



@grossws ping, this issue is waiting for your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by sanchox** [comment]
_From @ansibot on October 5, 2016 14:51_

@ansible, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by sanchox** [comment]
_From @phemmer on October 9, 2016 19:15_

See https://github.com/ansible/ansible/issues/16134

**Comment by sanchox** [comment]
_From @prupert on October 10, 2016 8:45_

> See ansible/ansible#16134

@phemmer, although related, this is another bug.

**Comment by sanchox** [comment]
_From @ansibot on October 21, 2016 8:36_

@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by sanchox** [comment]
_From @ansibot on November 6, 2016 9:35_

@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by sanchox** [comment]
_From @ansibot on November 22, 2016 9:43_

@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by sanchox** [comment]
_From @ansibot on December 7, 2016 19:1_

This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by sanchox** [comment]
##### COMPONENT NAME

systemd module
##### ANSIBLE VERSION

```
ansible 2.3.0.0
```
##### OS / ENVIRONMENT

Ubuntu 16.04.2 LTS
Debian 8.6
##### SUMMARY

Get this on host with Ubuntu 16.04.2 LTS and Ansible 2.3.0.0.
Destination is Debian 8.6 with systemd on BeagleBone Black.

**Comment by jctanner** [comment]
@sanchox The service module will call into the systemd module on the backend if systemd is detected on the system due to the action plugin code. That being said, the systemd module is a very light abstraction around the systemctl command. In your test scenario, are you able to remedy the problem with only systemctl commands?

needs_info

**Comment by ansibot** [comment]
@sanchox This issue is waiting for your response. Please respond or the issue will be closed.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by rustyx** [comment]
The command to remove stale/nonexistent services is:

```
systemctl reset-failed
```

**Comment by ansibot** [comment]
@sanchox This issue is waiting for your response. Please respond or the issue will be closed.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@sanchox You have not responded to information requests in this issue so we will assume it no longer affects you. If you are still interested in this, please create a new issue with the requested information.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)
